weave

```diff
@@ -244,10 +264,16 @@ elif [ "$1" = "run" -a -z "$IS_LOCAL" ] ; then
         DNS_ARGS=$(exec_remote dns-args "$@" || true)
     fi
     shift $(dns_arg_count "$@")
+    if [ "$1" = "--no-rewrite-hosts" ] ; then
```

Comment:
I think it is ok to fix the order for this. Otherwise we would have to do some tricks in order to parse the args for both the local `run`: 

    if [ "$1" != "--without-dns" ] ; then
        DNS_ARGS=$(exec_remote dns-args "$@" || true)
    fi
    shift $(dns_arg_count "$@")
    if [ "$1" = "--no-rewrite-hosts" ] ; then
        shift
    else
        extra_hosts_args "$@"
        DNS_EXTRA_HOSTS_ARGS="--rewrite-hosts $DNS_EXTRA_HOSTS_ARGS"
    fi

and the remote `run`

        dns_args "$@"
        shift $(dns_arg_count "$@")
        REWRITE_HOSTS=1
        if [ "$1" = "--no-rewrite-hosts" ] ; then
            REWRITE_HOSTS=
            shift
        fi